modular_citadel/code/modules/reagents/chemistry/reagents/fermi_reagents.dm

```diff
@@ -0,0 +1,1596 @@
+		M.reagents.remove_reagent(src.id, amount, FALSE)
+		M.reagents.add_reagent(src.InverseChem, amount, FALSE, other_purity = 1)
+		return
+	else
+		//var/pureVol = amount * puritys
+		var/impureVol = amount * (1 - purity) //turns impure ratio into impure chem
+		M.reagents.remove_reagent(src.id, (impureVol), FALSE)
+		M.reagents.add_reagent(src.ImpureChem, impureVol, FALSE, other_purity = 1)
+	return
+
+//When merging two fermichems, see above
+/datum/reagent/fermi/on_merge(data, amount, mob/living/carbon/M, purity)//basically on_mob_add but for merging
+	. = ..()
+	if(!ishuman(M))
+		return
+	if (purity < 0)
+		CRASH("Purity below 0 for chem: [src.id], Please let Fermis Know!")
+	if (purity == 1 || src.DoNotSplit == TRUE)
+		return
+	else if (src.InverseChemVal > purity)
+		M.reagents.remove_reagent(src.id, amount, FALSE)
+		M.reagents.add_reagent(src.InverseChem, amount, FALSE, other_purity = 1)
+		for(var/datum/reagent/fermi/R in M.reagents.reagent_list)
+			if(R.name == "")
+				R.name = src.name//Negative effects are hidden
+		return
+	else
+		var/impureVol = amount * (1 - purity)
+		M.reagents.remove_reagent(src.id, impureVol, FALSE)
+		M.reagents.add_reagent(src.ImpureChem, impureVol, FALSE, other_purity = 1)
+		for(var/datum/reagent/fermi/R in M.reagents.reagent_list)
+			if(R.name == "")
+				R.name = src.name//Negative effects are hidden
+	return
+
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+//										EIGENSTASIUM
+///////////////////////////////////////////////////////////////////////////////////////////////////
+//eigenstate Chem
+//Teleports you to chemistry and back
+//OD teleports you randomly around the Station
+//Addiction send you on a wild ride and replaces you with an alternative reality version of yourself.
+//During the process you get really hungry, then your items start teleporting randomly,
+//then alternative versions of yourself are brought in from a different universe and they yell at you.
+//and finally you yourself get teleported to an alternative universe, and character your playing is replaced with said alternative
+
+/datum/reagent/fermi/eigenstate
+	name = "Eigenstasium"
+	id = "eigenstate"
+	description = "A strange mixture formed from a controlled reaction of bluespace with plasma, that causes localised eigenstate fluxuations within the patient"
+	taste_description = "wiggly cosmic dust."
+	color = "#5020H4" // rgb: 50, 20, 255
+	overdose_threshold = 15
+	addiction_threshold = 15
+	metabolization_rate = 0.5 * REAGENTS_METABOLISM
+	addiction_stage2_end = 30
+	addiction_stage3_end = 41
+	addiction_stage4_end = 44 //Incase it's too long
+	var/location_created
+	//var/turf/open/location_created
+	var/turf/open/location_return = null
+	var/addictCyc1 = 0
+	var/addictCyc2 = 0
+	var/addictCyc3 = 0
+	var/addictCyc4 = 0
+	var/mob/living/fermi_Tclone = null
+	var/teleBool = FALSE
+	mob/living/carbon/purgeBody
```

Comment:
needs a var/ and same thing on GC applies: make sure that it doesn't interfere with gc/deletions.

Reply:
Deleted